### PR TITLE
List-my-note API

### PR DESCRIPTION
**Pull request by JamesCamel**

### CODE REVIEW

File: lib/routes.js [hunk 1]
```diff
@@ -70,6 +70,8 @@ appRouter.get('/s/:shortid/:action', response.publishNoteActions)
 appRouter.get('/p/:shortid', response.showPublishSlide)
 // publish slide actions
 appRouter.get('/p/:shortid/:action', response.publishSlideActions)
+// gey my note list
+appRouter.get('/myNotes', noteController.listMyNotes)
```
Comment (jackycute) [inline lib/routes.js:74]: Maybe we can use like `/api/notes` instead of root level path to avoid occupied more routes when introduce more APIs.

Comment (Yukaii) [reply lib/routes.js:74]: Yes we should do this. @JamesCamel Please change the route to `/api/notes/myNotes` 👍

File: lib/note/index.js [hunk 2]
```diff
@@ -190,6 +189,53 @@ async function noteActions (req, res) {
   }
 }
 
+async function getMyNoteList (userid, callback) {
```
Comment (Yukaii) [inline lib/note/index.js:192]:
```suggestion
async function getMyNoteList (userId, callback) {
```

File: lib/note/index.js [hunk 3]
```diff
@@ -190,6 +189,53 @@ async function noteActions (req, res) {
...
+    where: {
+      ownerId: userid
+    }
+  })
+  if (!myNotes) {
+    return callback(null, null)
+  }
+  try {
+    const myNoteList = []
+    for (let i = 0; i < myNotes.length; i++) {
```
Comment (Yukaii) [inline lib/note/index.js:203]:
```js
const myNoteList = myNotes.map(note => ({
        id: Note.encodeNoteId(note.id),
        text: note.title,
        tags: Note.parseNoteInfo(note.content).tags,
        createdAt: note.createdAt,
        lastchangeAt: note.lastchangeAt,
        shortId: note.shortid
}))
```

File: lib/routes.js (hunk 1 above)
Comment (JamesCamel) [reply lib/routes.js]: Fixed.

File: lib/note/index.js (hunk 2 above)
Comment (JamesCamel) [reply lib/note/index.js]: Fixed.

File: lib/note/index.js (hunk 3 above)
Comment (JamesCamel) [reply lib/note/index.js]: Fixed.

**Pull request merged**